akka-stream/src/main/scala/akka/stream/io/SynchronousFileSource.scala

```diff
@@ -25,8 +25,7 @@ object SynchronousFileSource {
    * It materializes a [[Future]] containing the number of bytes read from the source file upon completion.
    */
   def apply(f: File, chunkSize: Int = DefaultChunkSize): Source[ByteString, Future[Long]] =
-    new Source(new SynchronousFileSource(f, chunkSize, DefaultAttributes, Source.shape("SynchronousFileSource")))
-      .named(DefaultAttributes.nameOption.get)
+    new Source(new SynchronousFileSource(f, chunkSize, DefaultAttributes, Source.shape("SynchronousFileSource")).wrap()) // TO DISCUSS: I had to add wrap() here to make the name available
```

Comment:
Yes. So a module is either atomic or composite. It is the atomic modules that are turned into Publishers/Subscribers, the composite ones are just an organizational abstraction. Composite modules contain submodules, possibly recursively. All modules treat all unwired input and output ports as external ports.

 - grow(other: Module): takes the current module and adds the submodules of the other module to it (or, looking another way, it makes a new module which contains the submodules of both) without changing existing wiring between the submodules (i.e. if submodules were wired in one of the modules, they are still wired in the new module). This does not add a hierarchy level. Grow also can define how to calculate the materialized value of the resulting module from the materialized values of the original modules.
 - connect(Outport, Inport): takes a module, and "destroys" one external input and one external output port, wiring them together. So the number of external ports will be less by 2. This does not add a hierarchy level. Materialized values are not changed.
 - wrap(): Creates a new module that contains the original module as a submodule. This adds a hierarchy level. This also means that grow will now affect this new toplevel module, the structure of the original module is unchanged (since it is now a submodule). Materialized values are not changed.